File: frontend/src/app/shared/dto/tmdb/TmdbMovieDetailDto.ts
import {TmdbGenreDto} from "./TmdbGenreDto";
import {TmdbProductionCompanyDto} from "./TmdbProductionCompanyDto";
import {TmdbSpokenLanguagesDto} from "./TmdbSpokenLanguagesDto";

export interface TmdbMovieDetailDto {
  adult: boolean;
  backdrop_path: string;
  budget: number;
  genres: TmdbGenreDto[];
  homepage: string;
  id: number;
  imdb_id: string;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string;
  production_companies: TmdbProductionCompanyDto[];
  release_date: string;
  revenue: number;
  runtime: number;
  spoken_languages: TmdbSpokenLanguagesDto[];
  status: string;
  tagline: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}

File: src/main/java/ch/andreskonrad/torenta/tmdb/dto/TmdbMovieDetailDto.java
package ch.andreskonrad.torenta.tmdb.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@Getter
@JsonIgnoreProperties(ignoreUnknown = true)
@EqualsAndHashCode(of = "id")
public class TmdbMovieDetailDto {

    private boolean adult;
    private String backdrop_path;
    private int budget;
    private List<TmdbGenreDto> genres;
    private String homepage;
    private int id;
    private String imdb_id;
    private String original_language;
    private String original_title;
    private String overview;
    private double popularity;
    private String poster_path;
    private List<TmdbProductionCompanyDto> production_companies;
    private String release_date;
    private int revenue;
    private int runtime;
    private List<TmdbSpokenLanguagesDto> spoken_languages;
    private String status;
    private String tagline;
    private String title;
    private boolean video;
    private double vote_average;
    private int vote_count;
}
